lib/fs/filesystem.go

```diff
@@ -22,23 +22,39 @@ type Filesystem interface {
 	DirNames(name string) ([]string, error)
 	Lstat(name string) (FileInfo, error)
 	Mkdir(name string, perm FileMode) error
+	MkdirAll(name string, perm FileMode) error
 	Open(name string) (File, error)
+	OpenFile(name string, flags int, mode FileMode) (File, error)
 	ReadSymlink(name string) (string, error)
 	Remove(name string) error
+	RemoveAll(name string) error
 	Rename(oldname, newname string) error
 	Stat(name string) (FileInfo, error)
 	SymlinksSupported() bool
 	Walk(root string, walkFn WalkFunc) error
+	Show(name string) error
+	Hide(name string) error
+	Glob(pattern string) ([]string, error)
+	SyncDir(name string) error
+	Roots() ([]string, error)
+	Usage(name string) (Usage, error)
```

Comment:
I guess this is clever enough to check usage in a specific directory, but in practice it's never used.